_modules/vault.py

```diff
@@ -177,16 +177,32 @@ def cached_read(path, cache_prefix='', **kwargs):
             lease_valid = True
         else:
             lease_valid = False
-            vault_client.delete(cache_path)
-
-    if not vault_data or not lease_valid:
+            if not vault_data['renewable']:
+                vault_client.delete(cache_path)
+            else:
+                __salt__['event.send'](
+                    'vault/cache/renew/{0}'.format(cache_path),
+                    data={
+                        'message': 'The cached lease at {0} is renewable. It will be '
+                        'renewed and cached data will remain intact.'
+                        .format(cache_path)
+                    })
+                vault_client.renew_secret(vault_data['lease_id'])
+
+    if not vault_data or not (lease_valid and not vault_data['renewable']):
         __salt__['event.send'](
             'vault/cache/miss/{0}'.format(cache_path),
             data={
                 'message': 'The cached lease at {0} is either invalid or '
                 'expired. It will be regenerated and cached with new data.'
                 .format(cache_path)
             })
+    return cache_path, vault_client, vault_data
+
+
+def cached_read(path, cache_prefix='', **kwargs):
+    check_cached_lease(path, cache_prefix='', **kwargs)
```

Comment:
This isn't assigning any of the return values to local variables, so `vault_data` in the next line will be undefined.